cli/internal/reference/compref/compref.go

```diff
@@ -185,25 +185,70 @@ func Parse(input string) (*Ref, error) {
 		return nil, fmt.Errorf("invalid component name %q in %q, must match %q", ref.Component, originalInput, ComponentRegex)
 	}
 
-	// Step 6: Resolve type if not explicitly given
+	// Step 6: Build repository object using ParseRepository
+	var repositorySpec string
+	if ref.Type != "" {
+		repositorySpec = ref.Type + "::" + input
+	} else {
+		repositorySpec = input
+	}
+
+	repository, err := ParseRepository(repositorySpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse repository: %w", err)
+	}
+
+	ref.Repository = repository
+
+	// Extract the type from the parsed repository for consistency
 	if ref.Type == "" {
-		t, err := GuessType(input)
+		switch repository.(type) {
+		case *ociv1.Repository:
+			ref.Type = runtime.NewVersionedType(ociv1.Type, ociv1.Version).String()
+		case *ctfv1.Repository:
+			ref.Type = runtime.NewVersionedType(ctfv1.Type, ctfv1.Version).String()
+		}
+	}
+
+	return ref, nil
+}
+
+// ParseRepository parses a repository specification string and returns a typed repository object.
```

Comment:
again this is a repository ref and not a spec. a spec is a known nomen in OCM terminology and is the full spec yaml

Reply:
yes, I forgot to rename all occurrences. 🙇